### DISCUSSION TITLE
Working with mdc - how to temporarily change the pattern while the mdc is set?

### DISCUSSION THREAD

**Discussion by persona94** [gabime/spdlog#3087] [Q&A]
Looking at the example

```
void mdc_example()
{
    spdlog::mdc::put("key1", "value1");
    spdlog::mdc::put("key2", "value2");
    // if not using the default format, use the %& formatter to print mdc data
    // spdlog::set_pattern("[%H:%M:%S %z] [%^%L%$] [%&] %v");
}
```

If I do have a custom pattern then if I call `set_pattern` it will apply that pattern to log statements called from every thread even though the mdc is only set for 1 thread. So while the mdc is set on 1 thread, all threads will show an empty [] field. Is there any way to say "set this pattern for this thread only"?

**Comment by gabime** [comment]
No, there is no such way.
